flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -182,6 +182,36 @@ public Dispatcher(
 	// Lifecycle methods
 	//------------------------------------------------------
 
+	@Override
+	public void onStart() throws Exception {
+		try {
+			super.onStart();
+			startDispatcherServices();
+		} catch (Exception e) {
+			final DispatcherException exception = new DispatcherException(String.format("Could not start the Dispatcher %s", getAddress()), e);
+			onFatalError(exception);
+			throw exception;
```

Comment:
Yes, if `onFatalError` does not terminate the JVM for example. Since the `Dispatcher` does not know what will happen `onFatalError` I think it is safer to let the exception bubble up.